lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -7,6 +7,70 @@
 from ..population.dwelltime import DwelltimeModel
 
 
+def _extract_dwelltime_data_from_groups(groups, exclude_ambiguous_dwells):
+    """Compute data needed for dwelltime analysis from a dictionary of KymoTrackGroups.
+
+    This data consists of dwelltimes and observation limits per track. Note that dwelltimes of zero
+    are automatically dropped.
+
+    Parameters
+    ----------
+    groups : iterable of KymoTrackGroup
+        An iterable which provides a sequence of KymoTrackGroup. Note that each group can only
+        have one `Kymo` associated with it.
+    exclude_ambiguous_dwells : bool
+        Determines whether to exclude dwelltimes which are not exactly determined. If `True`,
+        tracks which start in the first frame or end in the last frame of the kymograph are not
+        used in the analysis, since the exact start/stop times of the binding event are not
+        definitively known.
+
+    Returns
+    -------
+    dwelltimes : numpy.ndarray
+        Dwelltimes
+    min_obs : numpy.ndarray
+        List of minimum observation times extracted from the kymos
+    max_obs : numpy.ndarray
+        List of maximum observation time
+    removed_zeros : bool
+        Whether zeroes were dropped
+
+    Raises
+    ------
+    ValueError
+        if one of the KymoTrackGroups has more than one `Kymo` associated with it
+    """
+    removed_zeros = False
+
+    def extract_dwelltime_data(group):
+        nonlocal removed_zeros
+
+        if len(group._kymos) > 1:
```

Comment:
We have no hard guidelines on touching private API that belongs to Pylake itself. We mostly decide this on a case-by-case basis.

While I'm aware that that is the ideal situation, in this case, the classes involved here are user-facing classes and we don't want to expose implementational details to the user.

We have quite a few of these functions that we consider "public" for internal use, but which are actually protected because we want to be able to change them at any time without the risk that outside users may end up depending on them. In addition, we also don't really want to pollute the namespace with them.